server/src/main/java/org/candlepin/model/AbstractHibernateObject.java

```diff
@@ -33,12 +33,17 @@
 
 /**
  * Abstract class for hibernate entities
+ *
+ * @param <T>
+ *  Entity type extending this class; should be the name of the subclass
  */
 @MappedSuperclass
 @XmlType(name = "CandlepinObject")
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonFilter("DefaultFilter")
-public abstract class AbstractHibernateObject implements Persisted, Serializable {
+public abstract class AbstractHibernateObject<T extends AbstractHibernateObject>
```

Comment:
What was the "covarience in the parameters" issue?  Is this an issue with the `populate` method calling super?